Bryce/add reporting docs (#14533)
DataDog/documentation: content/en/dashboards/reporting.md

---
title: Reporting
kind: documentation
disable_toc: false
is_public: true
beta: true
is_beta: true
further_reading:
  - link: "https://www.datadoghq.com/blog/template-variable-associated-values/"
    tag: "Blog"
    text: "Use associated template variables to refine your dashboards"
  - link: "https://learn.datadoghq.com/course/view.php?id=8"
    tag: "Learning Center"
    text: "Building Better Dashboards"
---

> <div class="alert alert-warning">This feature is currently in private beta. To request access, reach out to support@datadoghq.com.</div>

Reviewer: we're going to start rolling out next week (jul 25)-  we can actually add these docs as public (not unlisted) without the private beta disclaimer 🙌

## Overview

Reporting automatically sends a visual summary of a dashboard to selected recipients on a schedule. The generated reports display images of widgets from a dashboard in a linear HTML format.

> {{< img src="dashboards/reporting/report_example.png" alt="An example report titled Checkout KPI Report showing a date, a description, a dashboard link, and 3 images of dashboard widgets" style="width:100%;" >}}

Reviewer 2: The `+/-` icons on the Geomap bother me; on the one hand it's a faithful representation of reports, on the other hand it looks wrong and seeing it here makes me wish we had fixed this.

I'm going to see if I can take these icons out of the Geomap snapshots; if I can get this working I'll let you know & ask for a revised snapshot.

Reviewer 2: Took a look at this yesterday, was able to make a PR to fix it: https://github.com/DataDog/web-ui/pull/61225, currently in review.

## Schedule a report

> Create a new report from any dashboard by opening the dashboard’s cog menu and selecting **Schedule a Report**.

Reviewer 2: Regarding **any dashboard**:

- This feature is only supported for timeboards.
  - Does that mean we should use the word "timeboard"? I'm not sure -- there are 3 types of custom dashboard, and while 2 of them _(the non-screenboard options)_ are internally considered timeboards, only 1 is explicitly presented to users as such. See [here](https://docs.datadoghq.com/dashboards/) for an example of what I mean.
- We support it for custom timeboards and integration timeboards, but currently are only planning on GAing the custom timeboard support. So I think it would be best to specify **custom** here.

Reviewer 2: By the way, the documentation page linked above has an error on it: _"They are commonly used as status boards or storytelling view, which update in real-time and can represent fixed points in the past."_

Author: Thanks for confirming those details. From a docs perspective, I think it's best to err on the side of maximum accuracy, so I've updated this to "any custom timeboard".
Also, regarding the page linked above, I've opened the following PR to address this:
https://github.com/DataDog/documentation/pull/14566
Thanks for pointing it out!

The following widget types are supported:

- Change
- Distribution
- Geomap
- Group
- Heat Map
- Notes and Links
- Query Value
- Scatter Plot
- Service Summary
- Table
- Timeseries

> - Top List

Reviewer: can we add Monitor summary widget (https://docs.datadoghq.com/search/?s=monitor+summary+widget)? we just added support for this :)

In the configuration modal that opens, set a schedule for the report to determine when and how often the report is sent. Set a time frame to determine the range of time displayed in the resulting report. The report time frame can be different from the time frame displayed on the dashboard.

> Add recipients to your report by entering their email addresses. The email associated with your Datadog account is automatically added as a recipient. You can remove yourself as a recipient by hovering over your email and clicking the red **X**. 

Reviewer 2: Note that the X isn't actually red until the X itself is directly hovered over. I think it would be more accurate to describe the X as "the X next to the email address" (or something like that) instead of by color.

Author: Thanks, updated!

> {{< img src="dashboards/reporting/report_configuration_modal.png" alt="The configuration modal for an individual dashboard report, with sections to set a schedule, add recipients, and customize email. At the bottom of the modal is a radio button to link to the live dashboard and other buttons to edit template variables, send preview, cancel, and save" style="width:100%;" >}}

Reviewer 2: The "Link to the live dashboard" checkbox currently doesn't do anything, and is about to be removed -- I think it might be worth either waiting for this UI change, or manually editing the photo to not have it. (I'm a little worried that this might confuse customers.)

Reviewer 2: The reporting modals are also about to have a header text update, we'll be able to make more accurate screenshots in a day or so. _(I don't think this distinction really matters from a documentation perspective but I thought I'd bring it up.)_

> {{< img src="dashboards/reporting/report_configuration_modal.png" alt="The configuration modal for an individual dashboard report, with sections to set a schedule, add recipients, and customize email. At the bottom of the modal is a radio button to link to the live dashboard and other buttons to edit template variables, send preview, cancel, and save" style="width:100%;" >}}

Reviewer: https://a.cl.ly/ApuXmD68 updated screenshot with slight UI tweaks

{{< img src="dashboards/reporting/reporting_configuration_modal.png" alt="The configuration modal for reporting, with two reports displayed, each showing their titles, tags, recipients, frequency, an option to toggle the report on or off, and a button to edit the report. At the bottom is a button to add a new report and a done button" style="width:100%;" >}}

## Permissions

Only users with the **Dashboard Report Write** permission can generate a report. This permission is turned on by default for Admins and off for all other roles. 

Images generated in reports show all data regardless of granular read restrictions. We recommend limiting the report permissions to users who have no granular read restrictions on data. To grant a user the **Dashboard Report Write** permission, create a new role with the **Dashboards Report Write** permission turned on and assign the user to this role. Alternatively, assign the **Admin** role to this user. To learn more about managing roles and permissions, see our [account management documentation][1]. 

> {{< img src="dashboards/reporting/dashboard_permissions.png" alt="A screenshot of an individual user's permissions from within the organization settings page. The dashboards report write permission is highlighted under the dashboards section" style="width:100%;" >}}

Reviewer 2: The permission's description mentions it's in beta. This should be fine for now but once the feature is GA `#dashboards-backend` will update the description, at which point we'd be interested in using an updated screenshot here.

---
title: Reporting
kind: documentation
disable_toc: false
is_public: true
further_reading:
  - link: "https://www.datadoghq.com/blog/dashboard-sharing/"
    tag: "Blog"
    text: "Share Datadog dashboards securely with anyone" 
  - link: "https://www.datadoghq.com/blog/template-variable-associated-values/"
    tag: "Blog"
    text: "Use associated template variables to refine your dashboards"
  - link: "https://learn.datadoghq.com/course/view.php?id=8"
    tag: "Learning Center"

>     text: "Building Better Dashboards"

Reviewer 3: This is a link to the old learning center and 404's

Author: Thanks for catching that, updated!

---

## Overview

> Reporting automatically sends a visual summary of a dashboard to selected recipients on a schedule. The generated reports display images of widgets from a dashboard in a linear HTML format.

Reviewer 3: ```suggestion
Datadog Reporting automatically sends a visual summary of a dashboard to selected recipients on a schedule. The generated reports display images of widgets from a dashboard in a linear HTML format.
```

> {{< img src="dashboards/reporting/report_example.png" alt="An example report titled Checkout KPI Report showing a date, a description, a dashboard link, the name of the Datadog organization, and 3 images of dashboard widgets" style="width:70%;" >}}

Reviewer 3: ```suggestion
{{< img src="dashboards/reporting/report_example.png" alt="An example report titled Checkout KPI Report that includes a dashboard link and images of dashboard widgets" style="width:70%;" >}}
```

## Schedule a report

Create a new report from any dashboard with a grid-based or automatic layout. Click on the dashboard’s cog icon and select **Schedule a Report**.

The following widget types are supported:

- [Change][1]
- [Distribution][2]
- [Geomap][3]
- [Group][4]
- [Heat Map][5]
- [Monitor Summary][6]
- [Notes and Links][7]
- [Scatter Plot][8]
- [Service Summary][9]
- [Table][10]
- [Timeseries][11]
- [Top List][12]
- [Query Value][13]

### Set a schedule

In the configuration modal that opens, set a schedule for the report to determine when and how often the report is sent. Set a time frame to determine the range of time displayed in the resulting report. The report time frame can be different from the time frame displayed on the dashboard.

### Add recipients

Add recipients to your report by entering their email addresses. The email associated with your Datadog account is automatically added as a recipient. You can remove yourself as a recipient by hovering over your email and clicking the **X** that appears next to it.   

> **Note:** Enterprise and Pro accounts can send reports to recipients outside of their organizations. 

Reviewer 3: ```suggestion

**Note:** Enterprise and Pro accounts can send reports to recipients outside of their organizations. 
```

{{< img src="dashboards/reporting/report_configuration_modal.png" alt="The configuration modal for an individual dashboard report, with sections to set a schedule, add recipients, and customize email. At the bottom of the modal are buttons to edit template variables, delete report, send preview, cancel, and save" style="width:100%;" >}}

### Customize the report

> Finally, customize the report to provide recipients with more context or a tailored view. The optional description appears at the top of each report to provide more context on the dashboard.

Reviewer 3: ```suggestion
You can customize the report to provide recipients with more context or a tailored view. The optional description appears at the top of each report to provide more context on the dashboard.
```

{{< img src="dashboards/reporting/dashboard_cog_menu.png" alt="A view of the menu displayed when hovering over the cog icon in a dashboard, showing the option to configure reports" style="width:50%;" >}}

From the configuration modal that opens, you can pause an existing report or create a new report. To see and edit the details of an existing report, or delete the report, click **Edit**.

{{< img src="dashboards/reporting/reporting_configuration_modal.png" alt="The configuration modal for reporting, with two reports displayed, each showing their titles, tags, recipients, frequency, an option to toggle the report on or off, and a button to edit the report. At the bottom is a button to add a new report and a done button" style="width:100%;" >}}

## Permissions

Only users with the **Dashboard Report Write** permission can generate a report. This permission is turned on by default for Admins and off for all other roles. 

> Images generated in reports show all data regardless of granular read restrictions. We recommend limiting the report permissions to users who have no granular read restrictions on data. To grant a user the **Dashboard Report Write** permission, create a new role with the **Dashboards Report Write** permission turned on and assign the user to this role. Alternatively, assign the **Admin** role to this user. To learn more about managing roles and permissions, read our [account management documentation][14]. 

Reviewer 3: ```suggestion
Images generated in reports show all data regardless of granular read restrictions. Datadog recommends limiting the report permissions to users who have no granular read restrictions on data. To grant a user the **Dashboard Report Write** permission, create a new role with the **Dashboards Report Write** permission turned on and assign the user to this role. Alternatively, assign the **Admin** role to this user. To learn more about managing roles and permissions, read the [account management documentation][14]. 
```
